Problem 1.cpp

```diff
@@ -0,0 +1,50 @@
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include <string>
+#include <numeric>
+
+using namespace std;
+
+int main()
+{
+	vector<string> a;
+	string p;
+	do {
+		getline(cin, p);
+		a.push_back(p);
+	} while (p != "");
+	a.pop_back();
+
+	if (all_of(a.begin(), a.end(), [](string& i) {return (i.size() >= 4); })) {
+		cout << "ALL\n";
+	}
+	else if (any_of(a.begin(), a.end(), [](string& i) {return (i.size() >= 4); })) {
+		cout << "SOME\n";
+	}
+	else if (none_of(a.begin(), a.end(), [](string& i) {return (i.size() >= 4); })) {
+		cout << "NONE\n";
+	}
+
+	for_each(a.begin(), a.end(), [](string& i) {cout << i.back() << "\n"; });
+
+	cout << count_if(a.begin(), a.end(), [](string& i) {return count_if(i.begin(), i.end(), isdigit); }) << "\n";
+
+	transform(a.begin(), a.end(), a.begin(), [](string& i) {transform(i.begin(), i.end(), i.begin(), tolower); return NULL; });
```

Comment:
Don't modify the string by reference, create a new one and return it form your function.
Also NULL is a C construct and you shouldn't use it in C++. You should use `nullptr`, which denotes a pointer type. 
Have you run this code? I'm not sure why it would compile because it's expecting a return value of type string and you are returning 0.

Reply:
The code ran fine with NULL but broke when I replaced it with nullptr. I simply removed the whole return and it was fixed.

Reply:
The code ran fine with NULL, but broke when I either put in nullptr or removed the return altogether.